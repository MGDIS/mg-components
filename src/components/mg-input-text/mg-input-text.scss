:host {
  // An other way to display / hide characters left
  // Need to be improved using sr-only class exemple for accessibility
  mg-character-left,
  mg-help-text {
    display: none;
  }
  input:focus ~ mg-character-left,
  input:focus ~ mg-help-text {
    display: inline;
  }
}
