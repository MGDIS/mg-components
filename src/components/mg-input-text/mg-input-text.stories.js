export default {
  name: 'blu',
  component: 'mg-input-text',
  title: 'Components/mg-input-text',
  parameters: {
    docs: {
      source: {
        code: `<mg-input-text label="Label" required has-colon></mg-input-text>`,
      },
    },
  },
};

/**
 * 1. camelCase arguments must be written in the template, for exemple labelOnTop must be placed in the template as label-on-top={args.labelOnTop}
 * 2. boolean arguments with a default true value must be added like display-character-left={args.displayCharacterLeft ? 'true' : 'false'}
 */
const Template = args => (
  <mg-input-text
    {...args}
    label-on-top={args.labelOnTop}
    label-colon={args.labelColon}
    pattern-error-message={args.patternErrorMessage}
    display-character-left={args.displayCharacterLeft ? 'true' : 'false'}
    character-left-template={args.characterLeftTemplate}
    help-text={args.helpText}
  ></mg-input-text>
);

export const MgInputText = Template.bind({});
MgInputText.args = {
  value: '',
  reference: 'reference-id',
  name: 'input-name',
  label: 'Label',
  labelOnTop: false,
  labelColon: true,
  placeholder: 'placeholder',
  maxlength: 400,
  required: true,
  pattern: undefined,
  patternErrorMessage: undefined,
  disabled: false,
  readonly: false,
  tooltip: 'This is a tooltip',
  displayCharacterLeft: true,
  characterLeftTemplate: '{counter} characters left.',
  helpText: 'Help text with <a href="https://www.exemple.com">a link</a>, a <strong>bold text</strong> and <em>an italic one</em>.',
};
