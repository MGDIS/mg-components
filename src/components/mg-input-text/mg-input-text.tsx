import { Component, Event, Host, h, Prop, Watch, EventEmitter, State } from '@stencil/core';
import { createID } from '../../utils/utils';

@Component({
  tag: 'mg-input-text',
  styleUrl: 'mg-input-text.scss',
  shadow: true,
})
export class MgInputText {

  /**
   * Internal
   */
  private characterLeftReference = '';
  private helpTextReference = '';

  /**
   * Component value
   */
   @Prop({ mutable: true, reflect: true }) value: string;

  /**
   * Input reference used for the input ID (id is a reserved prop in Stencil.js)
   * If not set, an ID will be created
   */
  @Prop() reference?: string = createID('mg-input-text');

  /**
   * Input name
   * If not set the value equals the reference
   */
   @Prop() name?: string = this.reference;

  /**
   * Input label
   * Required
   */
  @Prop() label!: string;
  @Watch('label')
  validateLabel(newValue: string) {
    if (typeof newValue !== 'string' || newValue === '') {
      throw new Error('Label is a required prop')
    };
  }

  /**
   * Input placeholder
   */
  @Prop() placeholder: string;

  /**
   * Input max length
   */
   @Prop() maxlength: number = 400;

  /**
   * Define if input is required
   */
   @Prop() required: boolean;

   /**
   * Define if input is disabled
   */
  @Prop() disabled: boolean;

  /**
   * Define if component should display character left
   */
   @Prop() displayCharacterLeft: boolean = true;

  /**
   * Template to use for characters left sentence
   */
   @Prop() characterLeftTemplate: string;

   /**
   * Template to use for characters left sentence
   */
    @Prop() helpText: string;

  /**
   * Aria attributes that need to be added to the input :
   * - nbCharLeft
   * - help text
   * - tooltip ?
   * (label is already linked throught for/id)
   */
  @State() ariaDescribedby: string[] = [];

  /**
   * Emmited event when value change
   */
  @Event() changed: EventEmitter<string>


  /**
   *
   * @param event
   */
  handleChange(event) {
    this.value = event?.target?.value;
    this.changed.emit(this.value)
  }

  /**
   * Check if props are well configured on init
   * Add required aria attributes
   */
  componentWillLoad() {
    this.validateLabel(this.label);
    if(this.displayCharacterLeft) {
      this.characterLeftReference = `${this.reference}-character-left`;
      this.ariaDescribedby.push(this.characterLeftReference);
    }
    if(this.helpText) {
      this.helpTextReference = `${this.reference}-help-text`;
      this.ariaDescribedby.push(this.helpTextReference);

    }
  }

  /**
   * Render
   */
  render() {
    return (
      <Host>
        <mg-label reference={this.reference} required={this.required}>{this.label}</mg-label>
        <input
          type="text"
          id={this.reference}
          name={this.name}
          placeholder={this.placeholder}
          title={this.placeholder}
          value={this.value}
          maxlength={this.maxlength}
          disabled={this.disabled}
          required={this.required}
          aria-describedby={this.ariaDescribedby.join(' ')}
          onInput={(e) => this.handleChange(e)}
        />
        { this.displayCharacterLeft && <mg-character-left
          reference={this.characterLeftReference}
          characters={this.value}
          maxlength={this.maxlength}
          template={this.characterLeftTemplate}
        ></mg-character-left> }
        { this.helpText && <mg-help-text reference={this.helpTextReference} innerHTML={this.helpText}></mg-help-text> }
      </Host>
    );
  }
}
