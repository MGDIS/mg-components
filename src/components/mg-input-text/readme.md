# mg-input-text



<!-- Auto Generated Below -->


## Properties

| Property                | Attribute                 | Description                                                                                                   | Type      | Default                     |
| ----------------------- | ------------------------- | ------------------------------------------------------------------------------------------------------------- | --------- | --------------------------- |
| `characterLeftTemplate` | `character-left-template` | Template to use for characters left sentence                                                                  | `string`  | `undefined`                 |
| `disabled`              | `disabled`                | Define if input is disabled                                                                                   | `boolean` | `undefined`                 |
| `displayCharacterLeft`  | `display-character-left`  | Define if component should display character left                                                             | `boolean` | `true`                      |
| `label` _(required)_    | `label`                   | Input label Required                                                                                          | `string`  | `undefined`                 |
| `maxlength`             | `maxlength`               | Input max length                                                                                              | `number`  | `400`                       |
| `name`                  | `name`                    | Input name If not set the value equals the reference                                                          | `string`  | `this.reference`            |
| `placeholder`           | `placeholder`             | Input placeholder                                                                                             | `string`  | `undefined`                 |
| `reference`             | `reference`               | Input reference used for the input ID (id is a reserved prop in Stencil.js) If not set, an ID will be created | `string`  | `createID('mg-input-text')` |
| `required`              | `required`                | Define if input is required                                                                                   | `boolean` | `undefined`                 |
| `value`                 | `value`                   | Component value                                                                                               | `string`  | `undefined`                 |


## Events

| Event     | Description                     | Type                  |
| --------- | ------------------------------- | --------------------- |
| `changed` | Emmited event when value change | `CustomEvent<string>` |


## Dependencies

### Depends on

- [mg-label](../mg-label)
- [mg-character-left](../mg-character-left)

### Graph
```mermaid
graph TD;
  mg-input-text --> mg-label
  mg-input-text --> mg-character-left
  style mg-input-text fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
