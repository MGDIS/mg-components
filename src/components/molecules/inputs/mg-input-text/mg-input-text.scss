@use '../styles/mg-input.scss';
@use '../styles/mg-input-box.scss';
@use '../styles/mg-input-width.scss';
@use '../styles/mg-input-charleft.scss';
@use '../../../../styles/a11y.scss';

:host {
  display: block;
}

/**
* Icon in input
*/

.mg-input.mg-input--has-icon .mg-input__input {
  position: relative;
  & > mg-icon {
    // layout
    position: absolute;
    top: 50%;
    left: var(--mg-inputs-spacer);
    // text
    color: var(--mg-inputs-color);
    // other
    transform: translateY(-50%);
  }
  .mg-input__box {
    padding-left: calc(var(--mg-icon-regular-size) + var(--mg-inputs-spacer) * 2);
  }
}
