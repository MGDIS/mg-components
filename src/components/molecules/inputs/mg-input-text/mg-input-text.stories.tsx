import { h } from "@stencil/core";

export default {
  component: 'mg-input-text',
  title: 'Molecules/mg-input-text',
};

/**
 * 1. camelCase arguments must be written in the template, for exemple labelOnTop must be placed in the template as label-on-top={args.labelOnTop}
 * 2. boolean arguments with a default true value must be added like display-character-left={args.displayCharacterLeft ? 'true' : 'false'}
 */
const Template = args => (
  <mg-input-text
    {...args}
    label-on-top={args.labelOnTop}
    label-colon={args.labelColon}
    pattern-error-message={args.patternErrorMessage}
    display-character-left={args.displayCharacterLeft ? 'true' : 'false'}
    character-left-template={args.characterLeftTemplate}
    help-text={args.helpText}
  ></mg-input-text>
);

/**
 * Global use
 */

export const MgInputText = Template.bind({});

MgInputText.args = {
  // Global
  value: '',
  identifier: 'identifier',
  name: 'input-name',
  // Label
  label: 'Label',
  labelOnTop: false,
  labelColon: true,
  // Input
  placeholder: 'placeholder',
  maxlength: 400,
  required: true,
  disabled: false,
  readonly: false,
  pattern: undefined,
  patternErrorMessage: undefined,
  // Tooltip
  tooltip: 'This is a tooltip',
  // Nb Char Left
  displayCharacterLeft: true,
  characterLeftTemplate: '{counter} characters left.',
  // Help Text
  helpText: 'Help text with html <strong>bold</strong>, <em>italic</em>.',
};

// Set exemple code for component
MgInputText.parameters = {
  docs: {
    source: {
      code: `<mg-input-text
  value="${MgInputText.args.value}"
  label="${MgInputText.args.label}"
  tooltip="${MgInputText.args.tooltip}"
  help-text="${MgInputText.args.helpText}"
  required
  has-colon
></mg-input-text>`,
    },
  },
};

/**
 * Email
 */

export const Email = Template.bind({});

Email.args = {
  ...MgInputText.args,
  // remove feature to focus on pattern
  tooltip: '',
  required: false,
  labelColon: false,
  label: "Adresse email",
  // Add pattern Email rules
  maxlength: 100,
  pattern: '[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,15}',
  patternErrorMessage: 'L\'information saisie est incorrecte (exemple : prenom.nom@exemple.fr)',
  // Help Text
  helpText: 'exemple : prenom.nom@exemple.fr',
}

// Set exemple code for component
Email.parameters = {
  docs: {
    source: {
      code: `<mg-input-text
  value="${Email.args.value}"
  label="${Email.args.label}"
  maxlength="${Email.args.maxlength}"
  pattern="${Email.args.pattern}"
  pattern-error-message="${Email.args.patternErrorMessage}"
  help-text="${Email.args.helpText}"
></mg-input-text>`,
    },
  },
};

/**
 * Emails
 */

export const Emails = Template.bind({});

Emails.args = {
  ...Email.args,
  label: "Adresses email",
  // Add pattern Emails rules
  maxlength: 200,
  pattern: '([a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,15}(;)*)+',
  patternErrorMessage: 'L\'information saisie est incorrecte (exemple : prenom.nom@exemple.fr;prenom.nom.2@exemple.fr)',
  // Help Text
  helpText: 'exemple : prenom.nom@exemple.fr;prenom.nom.2@exemple.fr',
}

// Set exemple code for component
Emails.parameters = {
  docs: {
    source: {
      code: `<mg-input-text
  value="${Emails.args.value}"
  label="${Emails.args.label}"
  maxlength="${Emails.args.maxlength}"
  pattern="${Emails.args.pattern}"
  pattern-error-message="${Emails.args.patternErrorMessage}"
  help-text="${Emails.args.helpText}"
></mg-input-text>`,
    },
  },
};

/**
 * RNA
 */

export const RNA = Template.bind({});
RNA.args = {
  ...Emails.args,
  label: "RNA",
  // Add pattern RNA rules
  maxlength: 10,
  pattern: '(W[0-9]{1}[a-zA-Z0-9]{1}[0-9]{7})',
  patternErrorMessage: 'Le num√©ro RNA n\'est pas valide (exemple: W123456789)',
  // Help Text
  helpText: 'exemple : W123456789',
}

// Set exemple code for component
RNA.parameters = {
  docs: {
    source: {
      code: `<mg-input-text
  value="${RNA.args.value}"
  label="${RNA.args.label}"
  maxlength="${RNA.args.maxlength}"
  pattern="${RNA.args.pattern}"
  pattern-error-message="${RNA.args.patternErrorMessage}"
  help-text="${RNA.args.helpText}"
></mg-input-text>`,
    },
  },
};

/**
 * URL
 */

 export const URL = Template.bind({});
 URL.args = {
   ...Emails.args,
   label: "URL",
   // Add pattern URL rules
   maxlength: 200,
   pattern: 'https?:\/\/(?:www\.)?([-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b)*(\/[\/\d\w\.-]*)*(?:[\?])*(.+)*',
   patternErrorMessage: 'Le format du champ n\'est pas valide (exemple: https://www.exemple.fr)',
   // Help Text
   helpText: 'Exemple: https://www.exemple.fr',
 }

 // Set exemple code for component
 URL.parameters = {
   docs: {
     source: {
       code: `<mg-input-text
   value="${URL.args.value}"
   label="${URL.args.label}"
   maxlength="${URL.args.maxlength}"
   pattern="${URL.args.pattern}"
   pattern-error-message="${URL.args.patternErrorMessage}"
   help-text="${URL.args.helpText}"
 ></mg-input-text>`,
     },
   },
 };
