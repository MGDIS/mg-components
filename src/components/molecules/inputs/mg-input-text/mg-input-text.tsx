import { Component, Event, h, Prop, EventEmitter, State } from '@stencil/core';
import { MgInput } from '../MgInput';
import { createID, ClassList } from '../../../../utils/components.utils';
import { messages } from '../../../../locales';

@Component({
  tag: 'mg-input-text',
  styleUrl: 'mg-input-text.scss',
  shadow: true,
})
export class MgInputText {
  /************
   * Internal *
   ************/

  private classFocus = 'is-focused';
  private classError = 'is-not-valid';

  /**************
   * Decorators *
   **************/

  /**
   * Component value
   */
  @Prop({ mutable: true, reflect: true }) value: string;

  /**
   * Identifier is used for the element ID (id is a reserved prop in Stencil.js)
   * If not set, it will be created.
   */
  @Prop() identifier?: string = createID('mg-input-text');

  /**
   * Input name
   * If not set the value equals the identifier
   */
  @Prop() name?: string = this.identifier;

  /**
   * Input label
   * Required
   */
  @Prop() label!: string;

  /**
   * Define if label is displayed on top
   */
  @Prop() labelOnTop: boolean;

  /**
   * Define if label is visible
   */
  @Prop() labelHide: boolean = false;

  /**
   * Input placeholder.
   * It should be a word or short phrase that demonstrates the expected type of data, not a replacement for labels or help text.
   */
  @Prop() placeholder: string;

  /**
   * Input max length
   */
  @Prop() maxlength: number = 400;

  /**
   * Define if input is required
   */
  @Prop() required: boolean = false;

  /**
   * Define if input is readonly
   */
  @Prop() readonly: boolean = false;

  /**
   * Define if input is disabled
   */
  @Prop() disabled: boolean = false;

  /**
   * Define input pattern to validate
   */
  @Prop() pattern: string;

  /**
   * Define input pattern error message
   */
  @Prop() patternErrorMessage: string;

  /**
   * Add a tooltip message next to the input
   */
  @Prop() tooltip: string;

  /**
   * Define if component should display character left
   */
  @Prop() displayCharacterLeft: boolean = true;

  /**
   * Template to use for characters left sentence
   */
  @Prop() characterLeftTemplate: string;

  /**
   * Add a help text under the input, usually expected data format and example
   */
  @Prop() helpText: string;

  /**
   * Define input pattern to validate
   */
  @Prop({ mutable: true, reflect: true }) valid: boolean;

  /**
   * Define input pattern error message
   */
  @Prop({ mutable: true, reflect: true }) invalid: boolean;

  /**
   * Component classes
   */
  @State() classList: ClassList = new ClassList(['mg-input--text', 'mg-input--width-full']);

  /**
   * Error message to display
   */
  @State() errorMessage: string;

  /**
   * Emmited event when value change
   */
  @Event({ eventName: 'value-change' }) valueChange: EventEmitter<string>;

  /**
   * Handle input event
   * @param event
   */
  private handleInput = (event: InputEvent & { target: HTMLInputElement }) => {
    this.value = event.target.value;
    this.valueChange.emit(this.value);
  };

  /**
   * Handle focus event
   */
  private handleFocus = () => {
    this.classList.add(this.classFocus);
    this.classList = new ClassList(this.classList.classes);
  };

  /**
   * Handle blur event
   * @param event
   */
  private handleBlur = (event: FocusEvent) => {
    // Manage focus
    this.classList.delete(this.classFocus);
    this.classList = new ClassList(this.classList.classes);
    // Check validity
    this.checkValidity(event.target);
  };

  /**
   * Check if input is valid
   * @param element
   */
  private checkValidity(element) {
    const validity = element.checkValidity();
    // Set error message
    this.errorMessage = undefined;
    // Does not match pattern
    if (!validity && element.validity.patternMismatch) {
      this.errorMessage = this.patternErrorMessage;
    }
    // required
    else if (!validity && element.validity.valueMissing) {
      this.errorMessage = messages.errors.required;
    }

    // Set validity
    this.valid = validity;
    this.invalid = !validity;

    // Update class
    if (validity) {
      this.classList.delete(this.classError);
    } else {
      this.classList.add(this.classError);
    }
  }

  /**
   * Validate patern configuration
   */
  private validatePattern = () => {
    if (
      this.pattern &&
      typeof this.pattern === 'string' &&
      this.pattern !== '' &&
      (this.patternErrorMessage === undefined || typeof this.patternErrorMessage !== 'string' || this.patternErrorMessage === '')
    ) {
      throw new Error('<mg-input-text> prop "pattern" must be paired with the prop "patternErrorMessage"');
    }
  };

  /*************
   * Lifecycle *
   *************/

  /**
   * Check if component props are well configured on init
   */
  componentWillLoad() {
    this.validatePattern();
  }

  render() {
    return (
      <MgInput
        identifier={this.identifier}
        classList={this.classList}
        label={this.label}
        labelOnTop={this.labelOnTop}
        labelHide={this.labelHide}
        required={this.required}
        readonly={this.readonly}
        value={this.value}
        readonlyValue={undefined}
        tooltip={this.tooltip}
        displayCharacterLeft={this.displayCharacterLeft}
        characterLeftTemplate={this.characterLeftTemplate}
        maxlength={this.maxlength}
        helpText={this.helpText}
        errorMessage={this.errorMessage}
        isFieldset={false}
      >
        <input
          type="text"
          class="mg-input__box"
          value={this.value}
          id={this.identifier}
          name={this.name}
          placeholder={this.placeholder}
          title={this.placeholder}
          maxlength={this.maxlength}
          disabled={this.disabled}
          required={this.required}
          pattern={this.pattern}
          onInput={this.handleInput}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
        />
      </MgInput>
    );
  }
}
