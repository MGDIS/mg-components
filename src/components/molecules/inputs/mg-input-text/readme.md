# mg-input-text

<!-- Auto Generated Below -->


## Properties

| Property                | Attribute                 | Description                                                                                                                                    | Type      | Default                     |
| ----------------------- | ------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | --------- | --------------------------- |
| `characterLeftTemplate` | `character-left-template` | Template to use for characters left sentence                                                                                                   | `string`  | `undefined`                 |
| `disabled`              | `disabled`                | Define if input is disabled                                                                                                                    | `boolean` | `false`                     |
| `displayCharacterLeft`  | `display-character-left`  | Define if component should display character left                                                                                              | `boolean` | `true`                      |
| `helpText`              | `help-text`               | Template to use for characters left sentence                                                                                                   | `string`  | `undefined`                 |
| `identifier`            | `identifier`              | Identifier is used for the element ID (id is a reserved prop in Stencil.js) If not set, it will be created.                                    | `string`  | `createID('mg-input-text')` |
| `invalid`               | `invalid`                 | Define input pattern error message                                                                                                             | `boolean` | `undefined`                 |
| `label` _(required)_    | `label`                   | Input label Required                                                                                                                           | `string`  | `undefined`                 |
| `labelColon`            | `label-colon`             | Define if label has colon ":"                                                                                                                  | `boolean` | `false`                     |
| `labelOnTop`            | `label-on-top`            | Define if label is displayed on top                                                                                                            | `boolean` | `undefined`                 |
| `maxlength`             | `maxlength`               | Input max length                                                                                                                               | `number`  | `400`                       |
| `name`                  | `name`                    | Input name If not set the value equals the identifier                                                                                          | `string`  | `this.identifier`           |
| `pattern`               | `pattern`                 | Define input pattern to validate                                                                                                               | `string`  | `undefined`                 |
| `patternErrorMessage`   | `pattern-error-message`   | Define input pattern error message                                                                                                             | `string`  | `undefined`                 |
| `placeholder`           | `placeholder`             | Input placeholder. It should be a word or short phrase that demonstrates the expected type of data, not a replacement for labels or help text. | `string`  | `undefined`                 |
| `readonly`              | `readonly`                | Define if input is readonly                                                                                                                    | `boolean` | `false`                     |
| `required`              | `required`                | Define if input is required                                                                                                                    | `boolean` | `false`                     |
| `tooltip`               | `tooltip`                 | Add a tooltip message next to the input                                                                                                        | `string`  | `undefined`                 |
| `valid`                 | `valid`                   | Define input pattern to validate                                                                                                               | `boolean` | `undefined`                 |
| `value`                 | `value`                   | Component value                                                                                                                                | `string`  | `undefined`                 |


## Events

| Event         | Description                     | Type                  |
| ------------- | ------------------------------- | --------------------- |
| `valueChange` | Emmited event when value change | `CustomEvent<string>` |


## Dependencies

### Depends on

- [mg-label](../../../atoms/mg-label)
- [mg-tooltip](../../../atoms/mg-tooltip)
- [mg-icon](../../../atoms/mg-icon)
- [mg-character-left](../../../atoms/mg-character-left)

### Graph
```mermaid
graph TD;
  mg-input-text --> mg-label
  mg-input-text --> mg-tooltip
  mg-input-text --> mg-icon
  mg-input-text --> mg-character-left
  style mg-input-text fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
