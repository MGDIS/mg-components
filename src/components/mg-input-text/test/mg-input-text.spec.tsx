import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgInputText } from '../mg-input-text';

const getPage = (args) => newSpecPage({
  components: [MgInputText],
  template: () => (<mg-input-text {...args}></mg-input-text>),
});

describe('mg-input-text', () => {
  it('renders', async () => {
    const args = {label: 'label', reference:"reference"};
    const { root } = await getPage(args);
    expect(root).toMatchSnapshot();
  });
});
