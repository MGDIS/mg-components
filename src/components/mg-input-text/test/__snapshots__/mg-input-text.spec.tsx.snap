// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-input-text renders 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <mg-label reference="reference">
      label
    </mg-label>
    <input aria-describedby="reference-character-left" id="reference" maxlength="400" name="reference" type="text">
    <mg-character-left maxlength="400" reference="reference-character-left"></mg-character-left>
  </mock:shadow-root>
</mg-input-text>
`;
